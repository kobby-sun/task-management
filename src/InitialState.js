import _ from 'lodash';

export const appState = {
    loading: false,
    error: false,
    task: null,
    tasks: !_.isEmpty(sessionStorage.getItem('tasks')) ? JSON.parse(sessionStorage.getItem('tasks')) : [],
}