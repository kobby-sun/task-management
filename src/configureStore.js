import { createStore, compose } from 'redux';
import createReducer from './reducers';

export default function configureStore() {

    const composeEnhancers = process.env.NODE_ENV !== 'production' && typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
        ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
            shouldHotReload: false
        })
        : compose;

    const store = createStore(createReducer(), composeEnhancers());

    if (module.hot) {
        module.hot.accept('./reducers', () => {
            store.replaceReducer(createReducer());
        });
    }

    return store;
}
