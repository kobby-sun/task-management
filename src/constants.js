export const VIEW_TASK = 'TaskList/VIEW_TASK';
export const LOAD_TASKS = 'TaskList/LOAD_TASKS';
export const ADD_TASK = 'TaskList/ADD_TASK';
export const DELETE_TASK = 'TaskList/DELETE_TASK';
export const COMPLETE_TASK = 'TaskList/COMPLETE_TASK';
export const SET_PRIORITY = 'TaskList/SET_PRIORITY';

export const TASK_PRIORITY = [
    { label: 'Very Low', value: '-2', badge: 'light' },
    { label: 'Low', value: '-1', badge: 'info' },
    { label: 'Normal', value: '0', badge: 'secondary' },
    { label: 'High', value: '1', badge: 'primary' },
    { label: 'Very High', value: '2', badge: 'warning' }
]