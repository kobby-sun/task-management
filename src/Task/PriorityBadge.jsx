import React from "react";
import _ from 'lodash';
import * as consts from '../constants'

const PriorityBadge = ({ p }) => {
    const priority = _.find(consts.TASK_PRIORITY, x => x.value === p)
    if (!priority) return null
    return <span className={`badge badge-pill badge-${priority.badge}`}>{priority.label}</span>
}

export default PriorityBadge