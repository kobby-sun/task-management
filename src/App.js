import React from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import './App.css';
import { Add, View, List } from './Task'

function AppRouter() {
  return (
    <Router>

      <div className="container">
        <br />
        <h3>Task List</h3>
        <hr />

        <nav>
          <ul>
            <li>
              <Link to="/">Task List</Link>
            </li>
            <li>
              <Link to="/task/add/">Add Task</Link>
            </li>
          </ul>
        </nav>

        <br />

        <Route path="/" exact component={() => <List />} />
        <Route path="/tasks" exact component={() => <List />} />
        <Route path="/tasks/:filter" component={() => <List />} />
        <Route path="/task/add" component={() => <Add />} />
        <Route path="/task/view/:id" component={(props) => <View {...props} />} />
      </div>
    </Router>
  );
}

export default AppRouter;