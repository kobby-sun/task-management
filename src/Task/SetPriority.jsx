import React from "react";
import { Dropdown } from 'react-bootstrap';
import { connect } from 'react-redux';
import { setPriority } from '../actions';
import * as consts from '../constants'
import _ from 'lodash';
import PriorityBadge from './PriorityBadge'

const SetTaskPriority = ({ setPriority, task }) => {
    if (!task) return null
    return <Dropdown>
        <Dropdown.Toggle variant="light" id="dropdown-basic">
            <PriorityBadge p={task.taskPriority} />
        </Dropdown.Toggle>

        <Dropdown.Menu>
            {_.map(consts.TASK_PRIORITY, x => <Dropdown.Item onSelect={() => setPriority({ id: task.id, taskPriority: x.value })} active={task.taskPriority === x.value} key={x.value}><PriorityBadge p={x.value} /></Dropdown.Item>)}
        </Dropdown.Menu>
    </Dropdown>
}

const mapDispatchToProps = (dispatch) => ({
    setPriority: (task) => dispatch(setPriority(task))
});


const decorators = _.flow([
    connect(null, mapDispatchToProps)
]);


export default decorators(SetTaskPriority)