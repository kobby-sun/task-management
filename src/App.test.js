import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import _ from 'lodash';
import shortid from 'shortid'
import { appState } from './InitialState'
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import * as consts from './constants'
import { Dropdown } from 'react-bootstrap';

import SetPriority from './Task/SetPriority'
import PriorityBadge from './Task/PriorityBadge'

const InitialState = {
  app: appState
}

const mockStore = configureStore()

const task = { id: shortid.generate(), taskName: 'Test task', taskDesc: 'Test task description', taskPriority: '0' }


const wrapper = (component) => {
  const store = mockStore(InitialState)
  return <Provider store={store}>{component}</Provider>
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(wrapper(<App />), div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('set task priority', () => {
  it('should render snapshot', () => {
    const component = renderer.create(wrapper(<SetPriority />));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render empty content', () => {
    const component = mount(wrapper(<SetPriority />));

    expect(component.find('SetPriority').get(0)).toBeFalsy()
  });

  it('should render dropdown', () => {
    const component = mount(wrapper(<SetPriority task={task} />));

    expect(component.find('Dropdown').get(0)).toBeTruthy()
  });

  it('should render all priority options', () => {
    const component = mount(wrapper(<SetPriority task={task} />));

    expect(component.find(Dropdown.Menu)).toHaveLength(1)
    component.find(Dropdown.Toggle).simulate('click');
    expect(component.find(Dropdown.Item)).toHaveLength(consts.TASK_PRIORITY.length)
  });

  it('should render active priority option', () => {
    const component = mount(wrapper(<SetPriority task={task} />));

    expect(component.find(Dropdown.Menu)).toHaveLength(1)
    component.find(Dropdown.Toggle).simulate('click');
    expect(component.find(Dropdown.Menu).children().find(Dropdown.Item).first().find(PriorityBadge)).toHaveLength(1)
    expect(component.find(Dropdown.Menu).children().find(Dropdown.Item).children().find({ active: true }).first().find(PriorityBadge).prop('p')).toEqual(task.taskPriority)
  });

});