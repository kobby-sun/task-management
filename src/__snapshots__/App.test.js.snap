// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set task priority should render empty 1`] = `null`;

exports[`Set task priority should render snapshot 1`] = `null`;

exports[`set task priority should render snapshot 1`] = `null`;
